WordPress/gutenberg
# Send non-deprecated language tags from Android in initialProps

## Description

Fixes https://github.com/wordpress-mobile/gutenberg-mobile/issues/4368

By default, the Android/Java Locale class uses deprecated language codes for Hebrew and Indonesian in order to not break backwards compatibility. But Gutenberg uses the new language codes, which prevented the proper translations from being picked up for those languages on Android. This updates Android to instead send the new language codes to Gutenberg in the initial props.

I'm applying this fix on the Android native side because (1) it allows me to use `Locale::toLanguageTag` to "revert" the change in the language code. This felt right to me since it is the Locale class itself that was switching to the deprecated language code and causing this problem in the first place.

## How has this been tested?
**Note that this has to be tested via WPAndroid ([relevant PR](https://github.com/wordpress-mobile/WordPress-Android/pull/15743)) because language changes are not picked up in the demo app.**

1. Set Android's language to Hebrew or Indonesian [[instructions](https://support.google.com/accounts/answer/32047?hl=en&co=GENIE.Platform%3DAndroid)]
2. Clean install WordPress for Android (clean install avoids the issue that updating the device language has no effect if the user previously set a language within WPAndroid's App settings)
3. Create a new post
4. Open the Block Inserter and observe that the block names are translated

## Screenshots 

<img width="400" src="https://user-images.githubusercontent.com/4656348/146996859-9dfbc89b-2788-47c5-a25a-96da1c5aac15.png">

## Types of changes
Bugfix

## Checklist:
